src/code/audio_playback.c

```diff
@@ -149,11 +151,152 @@ void Audio_NoteDisable(Note* note) {
     note->noteSubEu.bitField0.asBitfields.finished = 0;
     note->playbackState.parentLayer = NO_LAYER;
     note->playbackState.prevParentLayer = NO_LAYER;
-    note->playbackState.adsr.adsrAction.adsrBits.bits0 = 0;
-    note->playbackState.adsr.fadeOutVel = 0;
+    note->playbackState.adsr.adsrAction.adsrBits.state = 0;
+    note->playbackState.adsr.current = 0;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/audio_playback/Audio_ProcessNotes.s")
+void Audio_ProcessNotes(void) {
+    s32 pad[2];
+    NoteAttributes *attributes;
+    NoteSubEu *noteSubEu2;
+    NoteSubEu *noteSubEu;
+    Note *note;
+    NotePlaybackState *playbackState;
+    Reverb reverb;
+    u8 bookOffset;
+    f32 scale;
+    s32 i;
+
+    for (i = 0; i < gAudioContext.gMaxSimultaneousNotes; i++) {
+        note = &gAudioContext.gNotes[i];
+        noteSubEu2 = &gAudioContext.gNoteSubsEu[gAudioContext.gNoteSubEuOffset + i];
+        playbackState = &note->playbackState;
+        if (playbackState->parentLayer != NO_LAYER) {
+            if ((u32) playbackState->parentLayer < 0x7fffffffU) {
```

Comment:
are the u32 cast and U at the end of the number needed? Id also make all the F's uppercase in the number.

Other than that, make sure to run format.sh (after installing clang-format and clang-tidy).
Rest looks good to me

Reply:
also our style generally is to have nothing in the args if it takes nothing, unless void is needed to match of course

Reply:
The u32 cast is necessary -- it's converting a pointer to an integer. The U at the end of the number silences a signed->unsigned conversion warning and clarifies that an unsigned comparison is happening.

I'm surprised to hear that `()` is preferred, seems like it could hide errors (and will confuse mips_to_c). Apparently there's 99 `()` and 45 `(void)` in the codebase.

Reply:
I didnt realize there was that much of a discrepancy, I guess we should agree on something soon and keep it consistent